src/display/canvas.js

```diff
@@ -1417,18 +1417,54 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
       return shadow(this, 'isFontSubpixelAAEnabled', enabled);
     },
 
-    showText: function CanvasGraphics_showText(glyphs) {
+    showText: function CanvasGraphics_showText(unmergedGlyphs) {
       var current = this.current;
       var font = current.font;
       if (font.isType3Font) {
-        return this.showType3Text(glyphs);
+        return this.showType3Text(unmergedGlyphs);
       }
 
       var fontSize = current.fontSize;
       if (fontSize === 0) {
         return;
       }
 
+      // If glyphs placed next to each other are of the same type,
+      // we can merge them and reduce calls to fillText/drawChar.
+      var glyphs = unmergedGlyphs.reduce((arr, glyph, index) => {
```

Comment:
I think `arr` is not really a descriptive name. Perhaps `glyphList` or something like that would be better?